docs/tutorial/macros.md

```diff
@@ -3,79 +3,142 @@ id: scala-3-macros
 title: Scala 3 Macros
 ---
 
-## Inline and macros
+[Inline methods][inline] provide us with a elegant technique for metaprogramming by performing some operations at compile time.
+However, sometimes inlining is not enough and we need more powerful way to analyze and synthesize programs at compile time.
+Macros enable us to do exactly this: treat **programs as data** and manipulate them.
 
-Let us start simple, we will define a macro that will compute `xⁿ` for a known values `x` and `n`.
 
+## Macros Treat Programs as Data
+With a macro, we can analyze the syntax tree of a program and generate syntax trees at compile time.
```

Comment:
We should not mention trees in this document. Trees should only be introduced in the Reflection document. We know `Expr` contains ASTs but that is irrelevant for the `Expr` abstraction.

We also want to make as many users not think in terms of trees as possible. This omission is quite deliberate as we have seen that if they learn about trees, they will just start doing unsafe and long code rather than use the safer and shorter alternatives in `Expr` land.

Reply:
Most of the text containing `tree`  should probably contain `expression` instead.